### PR TITLE
add logstash-core resources to classpath during development

### LINKED ISSUE #6235: Logger doesn't seems to log the argument hash anymore
When you start logstash with `--log.level debug` you don't see the hash passed to the logger's methods, you only see the message.

```
Sending Logstash's logs to /Users/ph/es/logstash/logs which is now configured via log4j2.properties
[2016-11-10T15:21:14,370][DEBUG][logstash.runner          ] -------- Logstash Settings (* means modified) ---------
[2016-11-10T15:21:14,376][DEBUG][logstash.runner          ] node.name: "sashimi"
[2016-11-10T15:21:14,376][DEBUG][logstash.runner          ] path.data: "/Users/ph/es/logstash/data"
[2016-11-10T15:21:14,377][DEBUG][logstash.runner          ] *config.string: "input { stdin {} } output { stdout {}}"
[2016-11-10T15:21:14,378][DEBUG][logstash.runner          ] config.test_and_exit: false
[2016-11-10T15:21:14,378][DEBUG][logstash.runner          ] config.reload.automatic: false
[2016-11-10T15:21:14,378][DEBUG][logstash.runner          ] config.reload.interval: 3
[2016-11-10T15:21:14,378][DEBUG][logstash.runner          ] metric.collect: true
[2016-11-10T15:21:14,379][DEBUG][logstash.runner          ] pipeline.id: "main"
[2016-11-10T15:21:14,379][DEBUG][logstash.runner          ] pipeline.workers: 4
[2016-11-10T15:21:14,379][DEBUG][logstash.runner          ] pipeline.output.workers: 1
[2016-11-10T15:21:14,380][DEBUG][logstash.runner          ] pipeline.batch.size: 125
[2016-11-10T15:21:14,380][DEBUG][logstash.runner          ] pipeline.batch.delay: 5
[2016-11-10T15:21:14,380][DEBUG][logstash.runner          ] pipeline.unsafe_shutdown: false
[2016-11-10T15:21:14,380][DEBUG][logstash.runner          ] path.plugins: []
[2016-11-10T15:21:14,381][DEBUG][logstash.runner          ] config.debug: false
[2016-11-10T15:21:14,381][DEBUG][logstash.runner          ] *log.level: "debug" (default: "info")
[2016-11-10T15:21:14,381][DEBUG][logstash.runner          ] version: false
[2016-11-10T15:21:14,382][DEBUG][logstash.runner          ] help: false
[2016-11-10T15:21:14,382][DEBUG][logstash.runner          ] log.format: "plain"
[2016-11-10T15:21:14,382][DEBUG][logstash.runner          ] http.host: "127.0.0.1"
[2016-11-10T15:21:14,382][DEBUG][logstash.runner          ] http.port: 9600..9700
[2016-11-10T15:21:14,383][DEBUG][logstash.runner          ] http.environment: "production"
[2016-11-10T15:21:14,383][DEBUG][logstash.runner          ] queue.type: "memory"
[2016-11-10T15:21:14,383][DEBUG][logstash.runner          ] queue.page_capacity: 262144000
[2016-11-10T15:21:14,384][DEBUG][logstash.runner          ] queue.max_events: 0
[2016-11-10T15:21:14,384][DEBUG][logstash.runner          ] slowlog.threshold.warn: -1
[2016-11-10T15:21:14,384][DEBUG][logstash.runner          ] slowlog.threshold.info: -1
[2016-11-10T15:21:14,385][DEBUG][logstash.runner          ] slowlog.threshold.debug: -1
[2016-11-10T15:21:14,385][DEBUG][logstash.runner          ] slowlog.threshold.trace: -1
[2016-11-10T15:21:14,385][DEBUG][logstash.runner          ] path.queue: "/Users/ph/es/logstash/data/queue"
[2016-11-10T15:21:14,385][DEBUG][logstash.runner          ] path.settings: "/Users/ph/es/logstash/config"
[2016-11-10T15:21:14,385][DEBUG][logstash.runner          ] path.logs: "/Users/ph/es/logstash/logs"
[2016-11-10T15:21:14,386][DEBUG][logstash.runner          ] --------------- Logstash Settings -------------------
[2016-11-10T15:21:14,596][DEBUG][logstash.agent           ] Agent: Configuring metric collection
[2016-11-10T15:21:14,611][DEBUG][logstash.instrument.periodicpoller.os] PeriodicPoller: Starting
[2016-11-10T15:21:14,613][DEBUG][logstash.instrument.periodicpoller.jvm] PeriodicPoller: Starting
[2016-11-10T15:21:14,684][FATAL][logstash.runner          ] An unexpected error occurred!
```

### PR DESCRIPTION
**Pull request by talevy**
Fixes #6235.

### CODE REVIEW

**Comment by talevy** [comment]
@colinsurprenant mind reviewing when you get the chance? thanks!

File: logstash-core/lib/logstash-core/logstash-core.rb [hunk 1]
```diff
@@ -9,10 +9,12 @@ module LogStash
 
 # local dev setup
 classes_dir = File.expand_path("../../../build/classes/main", __FILE__)
+resources_dir = File.expand_path("../../../build/resources/main", __FILE__)
 
-if File.directory?(classes_dir)
+if File.directory?(classes_dir) and File.directory?(resources_dir)
```
Comment (colinsurprenant) [inline logstash-core/lib/logstash-core/logstash-core.rb]: we should avoid using `and`, instead use `&&` for boolean operator

Comment (colinsurprenant) [inline logstash-core/lib/logstash-core/logstash-core.rb]: FYI http://www.virtuouscode.com/2010/08/02/using-and-and-or-in-ruby/

**Comment by colinsurprenant** [comment]
minor comment about boolean operator, otherwise LGTM

**Comment by colinsurprenant** [comment]
planning on merging this? can't wait to have my loggers logging! :D

**Comment by suyograo** [comment]
@talevy ping

**Comment by talevy** [comment]
yes. I will update tomorrow

**Comment by talevy** [comment]
I am going to hold off on some tests since they are rather complex and will have to be refactored after we change the build strategy when merging all of logstash java code into one gradle project

**Pull request closed without merging**